main.go

```diff
@@ -555,82 +572,168 @@ func HotelSearch(w http.ResponseWriter, req *http.Request) {
 		))
 	}
 
-	results, err := globalCluster.SearchQuery("hotels", qp, &gocb.SearchOptions{Limit: 100})
+	results, err := app.cluster.SearchQuery("hotels-index", qp, &gocb.SearchOptions{Limit: 100})
 	if err != nil {
+		app.logger.Printf("Error occurred performing hotel search query: %s", err)
 		writeJsonFailure(w, 500, err)
 		return
 	}
 
-	respData.Data = []jsonHotel{}
+	hotelCols := []string{
+		"country",
+		"city",
+		"state",
+		"address",
+		"name",
+		"description",
+		"type",
+	}
+	var respData jsonHotelSearchResp
 	for results.Next() {
-		res, _ := globalCollection.LookupIn(results.Row().ID, []gocb.LookupInSpec{
-			gocb.GetSpec("country", nil),
-			gocb.GetSpec("city", nil),
-			gocb.GetSpec("state", nil),
-			gocb.GetSpec("address", nil),
-			gocb.GetSpec("name", nil),
-			gocb.GetSpec("description", nil),
-		}, nil)
-		// We ignore errors here since some hotels are missing various
-		//  pieces of data, but every key exists since it came from FTS.
+		res, err := hotels.Get(results.Row().ID, &gocb.GetOptions{
+			Project: hotelCols,
+		})
+		if err != nil {
+			app.logger.Printf("Error occurred fetching hotel: %s", err)
+			writeJsonFailure(w, 500, err)
+			return
+		}
 
 		var hotel jsonHotel
-		res.ContentAt(0, &hotel.Country)
-		res.ContentAt(1, &hotel.City)
-		res.ContentAt(2, &hotel.State)
-		res.ContentAt(3, &hotel.Address)
-		res.ContentAt(4, &hotel.Name)
-		res.ContentAt(5, &hotel.Description)
+		err = res.Content(&hotel)
+		if err != nil {
+			app.logger.Printf("Error occurred parsing hotel: %s", err)
+			writeJsonFailure(w, 500, err)
+			return
+		}
 		respData.Data = append(respData.Data, hotel)
 	}
+	b, err := json.Marshal(qp)
```

Comment:
Find it difficult to tell what `b` represents, maybe we could change to a more explicit variable name?